test/unittest/layers/layers_golden_tests.cpp

```diff
@@ -157,36 +158,38 @@ static void compareRunContext(RunLayerContext &rc, std::ifstream &file) {
       auto answer = tensor.clone();
       sizeCheckedReadTensor(answer, file, name + " at " + std::to_string(i));
 
-      if (name == "initial_weights") {
+      if (skip_compare) {
         continue;
       }
       EXPECT_EQ(tensor, answer) << name << " at " << std::to_string(i);
     }
   };
 
-  auto always = [](unsigned idx) { return true; };
-  auto only_trainable = [&rc](unsigned idx) {
+  auto always_read = [](unsigned idx) { return true; };
+  auto only_read_trainable = [&rc](unsigned idx) {
     return rc.weightHasGradient(idx);
   };
 
+  constexpr bool skip_compare = true;
+
   compare_tensors(rc.getNumWeights(),
-                  [&rc](unsigned idx) { return rc.getWeight(idx); }, always,
-                  "initial_weights");
+                  [&rc](unsigned idx) { return rc.getWeight(idx); },
+                  always_read, skip_compare, "initial_weights");
   compare_tensors(rc.getNumInputs(),
-                  [&rc](unsigned idx) { return rc.getInput(idx); }, always,
-                  "inputs");
+                  [&rc](unsigned idx) { return rc.getInput(idx); }, always_read,
+                  !skip_compare, "inputs");
```

Comment:
check the layer for `supportInPlace`. If it's true, skip checking input as it would be overwritten, or create two modes to configure it and check for both (maybe in later PR).

Reply:
if the layer is set to operate in `in-place`, then the layer/model need not perform such checks.
I guess this will be clarified with https://github.com/nnstreamer/nntrainer/blob/de0ae0d01127ef62afda12caa2185d396b27e138/nntrainer/layers/layer_context.h#L279-L284. 

> Shouldn't we have to make sure supportInPlace must not override the input if underlying memory does not clashing each other?

`context.executeInPlace()` tells the layer if the layer is expected to run in-place. So, this test can start with execute-in-place to be always false, but must test the layers which `support in-place` operations with execute-in-place set to true as well.

Reply:
Awesome, I'll make sure to make one in later PR